VMT/KCSA/KCSA-CVE-2020-2026.md

```diff
@@ -0,0 +1,98 @@
+can trick the kata runtime into mounting the untrusted container filesystem
+ on any host path, potentially allowing for code execution on the host. 
+This issue affects: Kata Containers 1.11 versions earlier than 1.11.1; 
+Kata Containers 1.10 versions earlier than 1.10.5. 
+
+affected-components:
+
+  - components: `kata-runtime`
+    version: Before v1.11.1, v1.10.5
+
+vulnerabilities:
+
+  - CVE-ID: CVE-2020-2026
+
+reporters:
+
+  - name: `Yuval Avrahami`
+    affiliation: `Palo Alto Networks`
+    reported:
+      - CVE-2020-2026
+
+issues:
+
+  links:
+    - https://github.com/kata-containers/runtime/issues/2712
+
+ reviews:
+
+  v1.11.0:
+    - https://github.com/kata-containers/runtime/pull/2713
+
+  type: GitHub
+
+reproduce:
+  - A malicious guest compromised before a container creation (e.g. a malicious 
+    guest image or a guest running multiple containers) can trick the kata 
+    runtime into mounting the untrusted container filesystem on any host path.
+
+    When Kata Containers is configured with overlay2 as storage driver, 
+    a malicious guest can create a symlink in the shared directory at 
+    `/run/kata-containers/shared/containers/${ctrid}/rootfs` to a target directory
+    on the host. Upon container creation, the kata runtime will be tricked into 
+    bind mounting the container filesystem at the target directory on the host.
+
+    To create the symlink the guest must know the container id as it’s a part 
+    of the mount’s target path. The first container in a sandbox is unique in that 
+    regard since its id is the sandbox id, which is known to the guest at the time of the mount.
+    If a guest is compromised before the first container is added to it (e.g. a malicious guest image), 
+    it can execute the following attack in case of overlay file systems:
+    1. The malicious guest receives the CreateSandbox message and extracts the sandbox 
+       id from it. The first container id matches the sandbox id and is derived from that message.
+    2. The malicious guest creates the malicious symlink at the shared directory, 
+       at /run/kata-containers/shared/containers/${first_ctrid}/rootfs
+    3. The malicious guest returns a response for CreateSandbox
+    4. Once the kata runtime on the host receives the CreateSandbox response, it tries to 
+       bind mount the container image at /run/kata-containers/shared/sandbox/$sbx_id/${first_ctrid}/rootfs
+    5. The malicious symlink redirects the mount operation to the target on the host.
+
+    In case of non initial container images, the guest does not know the container id.
+    In case of volume mounts and mounts related to platform-specific files such as
+    `/etc/hosts`, the mounts are performed in the shared directory on paths that are not
+    known to the guest. The guest would need to win a race condition between
+    `ensureDestinationExists` function call in the agent that creates the file or
```

Comment:
This function is [called](https://github.com/kata-containers/runtime/blob/ebe9677f23b574c5defacf57456d221d8ce901f2/virtcontainers/mount.go#L276) by the `kata-runtime`.